src/module.ts

```diff
@@ -0,0 +1,78 @@
+import { PanelConfig } from './panel-config';
+import { PanelCtrl } from 'grafana/app/plugins/sdk';
+import * as _ from 'lodash';
+import * as axios from 'axios/dist/axios';
+import './css/template.css';
+import { AC2Map } from './ac2-map';
+
+class Ctrl extends PanelCtrl {
+  static templateUrl = "partials/template.html";
+
+  private panelDefaults = {
+    host: "localhost",
+    port: "7890",
+    application: "Servers,1.0.0",
+    user: "admin",
+    password: "admin"
+  };
+
+  private setupDiagramTimer?: number = undefined;
+  private _panelConfig: PanelConfig;
+  private map: AC2Map;
+
+  constructor($scope: any, $injector) {
+    super($scope, $injector);
+    this._panelConfig = new PanelConfig(this.panel);
+    _.defaultsDeep(this.panel, this.panelDefaults);
+    this.events.on('init-edit-mode', this._onInitEditMode.bind(this));
+    this.map = new AC2Map();
+  }
+
+  restCall() {
+    const urlBase = `http://${this.panel.host}:${this.panel.port}`;
+    return axios({
+      method: 'POST',
+      url: `${urlBase}/api/Token`,
+      data: {
+        "User": this.panel.user,
+        "Password": this.panel.password
+      },
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      },
+    })
+      .then(response => {
+        return axios.get(`${urlBase}/api/Application?application=${this.panel.application}&api_key=${response.data}`);
+      });
+  }
+
+  setupDiagram() {
+    this.restCall()
+      .then(response => this.map.update(response))
+      .catch(error => {
+        console.error(error);
+        const container = document.getElementById("container");
```

Comment:
The error handling is a bit low level (getElementIdById + innerHTML)
As grafana extensions are based on angular, it should be a better way to do that.
You can create a github issue to fix that in another PR

Reply:
done